libdevcore/Exceptions.cpp

```diff
@@ -24,12 +24,14 @@ using namespace dev;
 
 char const* Exception::what() const noexcept
 {
+	// Return the comment if available.
 	if (string const* cmt = comment())
-		return cmt->c_str();
-	else
-		/// Boost accepts nullptr, but the C++ standard doesn't
-		/// and crashes on some platforms.
-		return std::exception::what();
+		return cmt->data();
```

Comment:
Actually it turned out we have a ton of places using `c_str`. Should we replace all?

Reply:
No. They are doing exactly the same. In C++11 the `data()` also returns null-terminated C-string. So `data()` is nicer, but I don't see reason to replace all `c_str()`s.